cpp/src/arrow/compute/row/row_encoder_internal.h

```diff
@@ -259,6 +334,9 @@ class ARROW_EXPORT RowEncoder {
   Status EncodeAndAppend(const ExecSpan& batch);
   Result<ExecBatch> Decode(int64_t num_rows, const int32_t* row_ids);
 
+  // Returns the encoded representation of the row at index i.
+  // If i is kRowIdForNulls, it returns the pre-encoded all-nulls
+  // row.
```

Comment:
Another optimization might be `std::string_view unsafe_encoded_row(int32_t i)`, which not copying the row. When the std::string cannot applying SSO, it would be benifit from less heap allocation

Reply:
Can you open a separate PR for that?

Reply:
> Can you open a separate PR for that?

Would do after this merged

Reply:
https://github.com/apache/arrow/issues/43909